docs/tutorials/timeLock.md

```diff
@@ -0,0 +1,80 @@
+---
+sidebar_position: 2
+---
+
+# Tutorial 2: Time-Lock
+
+## Overview
+
+In this tutorial, we will go over how to create a smart contract, which has a public method, that can only be unlocked once a certain point in time has passed.
+
+### What is a time-lock?
+
+In the context of smart contracts, a time-lock is a feature that restricts the spending of specific bitcoins until a specified future time or block height is reached. sCrypt offers capabilities to implement these types of time-locks in your smart contracts, providing a mechanism to ensure a transaction won't be included in a block before a certain point in time or block height is reached. In other words, the smart contract's method cannot be successfully invoked until that point in time has passed.
+
+### Implementation
+
+In sCrypt, a time-lock can be enforced by constraining the `locktime` and `sequence` values of the [script execution context](../how-to-write-a-contract/scriptcontext). This context pertains to the execution of the transaction, which includes a call to the smart contract's public method. Thus, if the value is constrained – for example, the `locktime` needs to be above the value `1690236000` (a Unix timestamp) – then this transaction cannot be included into the blockchain until that point in time.
+
+Note that the value of `locktime` can either be a Unix timestamp or a block height. For this value to be enforced, `sequence` also needs to be set to a value less than `0xffffffff`.
+
+## Contract properties
+
+Let's declare the properties of our smart contract:
+
+```ts
+// Any value of locktime, that is less than the value below, is 
+// considered to be a block height. Otherwise it is considered to be 
+// an Unix timestamp.
+static readonly LOCKTIME_BLOCK_HEIGHT_MARKER = 500000000
```

Comment:
Explain this number and link to reference like  https://wiki.bitcoinsv.io/index.php/NLocktime_and_nSequence. Move explanation outside of code.